conda_lock/conda_lock.py

```diff
@@ -734,12 +769,36 @@ def lock(
     default="",
 )
 @click.option("--auth-file", help="Path to the authentication file.", default="")
+@click.option(
+    "--validate-platform",
+    is_flag=True,
+    default=True,
+    help="Whether the platform compatibility between your lockfile and the host system should be validated.",
+)
 @click.argument("lock-file")
-def install(conda, mamba, micromamba, prefix, name, lock_file, auth, auth_file):
+def install(
+    conda,
+    mamba,
+    micromamba,
+    prefix,
+    name,
+    lock_file,
+    auth,
+    auth_file,
+    validate_platform,
+):
     """Perform a conda install"""
     auth = json.loads(auth) if auth else read_json(auth_file) if auth_file else None
     _conda_exe = determine_conda_executable(conda, mamba=mamba, micromamba=micromamba)
     install_func = partial(do_conda_install, conda=_conda_exe, prefix=prefix, name=name)
+    if validate_platform:
+        lockfile = read_file(lock_file)
+        try:
+            do_validate_platform(lockfile)
+        except RuntimeError as error:
+            raise RuntimeError(
+                error.args[0] + " Disable validation with `--validate-platform=False`."
+            )
```

Comment:
A bit hacky, we could make this a bit more readable with a custom error class `ValidationError`.